release_bot/github.py

```diff
@@ -13,16 +13,82 @@
 # You should have received a copy of the GNU General Public License
 # along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
-from os import listdir
+import os
+import time
 import re
 from tempfile import TemporaryDirectory
 from zipfile import ZipFile
-import requests
 
 from release_bot.git import Git
 from release_bot.exceptions import ReleaseException, GitException
 from release_bot.utils import insert_in_changelog, parse_changelog, look_for_version_files
 
+import jwt
+import requests
+
+
+# github app auth code "stolen" from https://github.com/swinton/github-app-demo.py
+class JWTAuth(requests.auth.AuthBase):
+    def __init__(self, iss, key, expiration=10 * 60):
+        self.iss = iss
+        self.key = key
+        self.expiration = expiration
+
+    def generate_token(self):
+        # Generate the JWT
+        payload = {
+          # issued at time
+          'iat': int(time.time()),
+          # JWT expiration time (10 minute maximum)
+          'exp': int(time.time()) + self.expiration,
+          # GitHub App's identifier
+          'iss': self.iss
+        }
+
+        tok = jwt.encode(payload, self.key, algorithm='RS256')
+
+        return tok.decode('utf-8')
+
+    def __call__(self, r):
+        r.headers['Authorization'] = 'bearer {}'.format(self.generate_token())
+        return r
+
+
+class GitHubApp:
+    def __init__(self, app_id, private_key_path):
+        self.session = requests.Session()
+        self.session.headers.update(dict(
+            accept='application/vnd.github.machine-man-preview+json'))
+        self.private_key = None
+        self.private_key_path = private_key_path
```

Comment:
@TomasTomecek You can actually save it in openshift secrets accessible via env var :)

Reply:
Yep, we can do that. I'm running the pod as rootless podman container now, so for me it's good enough to have the path stored in a config. But for future, we should utilize openshift secrets, that's a good point.

Reply:
That's what I did in `ansiwatch-bot`, while `check-in` still supports loading the key from a file.